lineapy/plugins/taskgen.py

```diff
@@ -0,0 +1,221 @@
+"""
+This taskgen files contains helper functions to create tasks.
+"""
+from typing import Dict
+
+from lineapy.graph_reader.artifact_collection import ArtifactCollection
+from lineapy.graph_reader.node_collection import ArtifactNodeCollection
+from lineapy.plugins.session_writers import BaseSessionWriter
+from lineapy.plugins.task import DagTaskBreakdown, TaskDefinition
+from lineapy.plugins.utils import load_plugin_template
+
+
+def get_task_definitions(
+    artifact_collection: ArtifactCollection,
+    pipeline_name: str,
+    task_breakdown: DagTaskBreakdown,
+) -> Dict[str, TaskDefinition]:
+    """
+    get_task_definitions returns a dictionary of TaskDefinitions
+
+    This function breaks down the artifact_collection into tasks based on the
+    task_breakdown parameter. This will give the main bulk of tasks that should
+    be included in a pipeline dag file.
+
+    Returns a `task_definitions` dictionary, which maps a key corresponding to the task name to
+    Linea's TaskDefinition object.
+    Specific framework implementations of PipelineWriters should serialize the TaskDefinition
+    objects to match the format for pipeline arguments that is expected by that framework.
+    """
+    if task_breakdown == DagTaskBreakdown.TaskAllSessions:
```

Comment:
Just want to confirm. This session is not accessible from our codebase right?

Reply:
Yeah, this needs to map to a dag_flavor for each framework. The reason I want to hide this is because each DAG has a different word for "task" so this concept of a task should not be exposed directly to users.